File: objc/ChainOfResponsibility/ChainOfResponsibility/MoneyDivison.m
#import "MoneyDivison.h"

@implementation MoneyDivison

-(void)handle:(Request*)request {
    if(request.money > 0)
        [self processRequest];
}

-(void)processRequest {
    NSLog(@"This request is processed by -- MoneyDivison");
}

@end

File: swift/ChainOfResponsibility/ChainOfResponsibility/MoneyDivison.swift
import Foundation

public class MoneyDivison : Divison {
    
    override func handle(_ request: Request) -> Void {
        if request.money > 0  {
            processRequest()
        }
    }
    
    override func processRequest() -> Void {
        print("This request is processed by -- MoneyDivison")
    }
}
